go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1432,6 +1441,11 @@ func (tsv *TabletServer) execRequest(
 		span.Annotate("shard", target.Shard)
 		span.Annotate("keyspace", target.Keyspace)
 	}
+	workloadName := sqlparser.UnspecifiedWorkloadName
+	if options != nil {
+		workloadName = options.WorkloadName
+	}
+	span.Annotate("workload_name", workloadName)
 	defer span.Finish()
```

Comment:
The reason I would like to add it here is because tracing tools (e.g. Honeycomb.io) don't allow you to analyze data from a given span joining it with data from parent spans. E.g. if I want to get in Honeycomb the amount of time spent serving queries grouping by workload name (where workload name is an attribute in the root span), basically it can't be done. You can only do group by attributes of the same span. Hence why annotating the child span of the DB query with the workload name can be useful.

Do you see a downside to it that perhaps I am missing?